test/set.e2e-spec.ts

```diff
@@ -1,443 +1,765 @@
-    it('/set/:id (PATCH)', async () => {
-      await request(app.getHttpServer())
-        .patch(`/api/set/${wrongId}`)
-        .set('Authorization', `Bearer ${token}`)
-        .expect(HttpStatus.NOT_FOUND)
-    })
+        it('/set/:id (DELETE) by id with hard delete by admin', async () => {
+            fakeAuthGuard.setUser(getMockAuthAdmin())
+            await request(app.getHttpServer())
+                .delete(`/set/${getSetSetupData()._id}?type=hard`)
+                .expect(HttpStatus.NO_CONTENT)
+            expect((await setModel.find()).length).toBe(0)
+        })
 
-    /*----------------------\ 
-    |         Tasks         |
-    \----------------------*/
-
-    it('/set/:id/task (POST)', async () => {
-      const res = await request(app.getHttpServer())
-        .post(`/api/set/${exampleSet._id}/task`)
-        .set('Authorization', `Bearer ${token}`)
-        .send({
-          type: exampleTask.type,
-          currentPlayerGender: exampleTask.currentPlayerGender,
-          message: exampleTask.message
-        })
-        .expect(HttpStatus.CREATED)
-      exampleTask._id = res.body._id
-
-      // Check if the count has been updated
-      const res2 = await request(app.getHttpServer())
-        .get(`/api/set/${exampleSet._id}`)
-        .set('Authorization', `Bearer ${token}`)
-        .expect(HttpStatus.OK)
-      expect(res2.body.truthCount === 1).toBeTruthy()
-      expect(exampleTask).toEqual(res.body)
-    })
+        // Negative test
+        it('/set/:id (DELETE) by id by wrong user', async () => {
+            fakeAuthGuard.setUser({
+                ...getMockAuthUser(),
+                userId: getWrongId(),
+            })
+            await request(app.getHttpServer())
+                .delete(`/set/${getSetSetupData()._id}`)
+                .expect(HttpStatus.NOT_FOUND)
+        })
 
-    // Negative test
-    it('/set/:id/task (POST)', async () => {
-      await request(app.getHttpServer())
-        .post(`/api/set/${wrongId}/task`)
-        .set('Authorization', `Bearer ${token}`)
-        .expect(HttpStatus.BAD_REQUEST)
-    })
+        // Negative test
+        it('/set/:id (DELETE) by wrong id', async () => {
+            await request(app.getHttpServer())
+                .delete(`/set/${getWrongId()}`)
+                .expect(HttpStatus.NOT_FOUND)
+        })
 
-    it('/set/:id/task/:taskid (PUT)', async () => {
-      const res = await request(app.getHttpServer())
-        .put(`/api/set/${exampleSet._id}/task/${exampleTask._id}`)
-        .set('Authorization', `Bearer ${token}`)
-        .send({
-          type: TaskType.DARE,
-          currentPlayerGender: exampleTask.currentPlayerGender,
-          message: exampleTask.message
-        })
-        .expect(HttpStatus.OK)
-      expect(res.body.daresCount === 1).toBeTruthy()
-      expect(res.body.truthCount === 0).toBeTruthy()
-    })
+        // Negative test
+        it('/set/:id (DELETE) by id with hard delete by admin', async () => {
```

Comment:
this name does not sound like a negative test

Reply:
Fixed